### ISSUE TITLE
CVPR'2017 iCaRL: Incremental Classifier and Representation Learning

### ISSUE THREAD

**Issue by swagshaw** [swagshaw/ResearchReading#5]
https://openaccess.thecvf.com/content_cvpr_2017/papers/Rebuffi_iCaRL_Incremental_Classifier_CVPR_2017_paper.pdf

**Comment by swagshaw** [comment]
## background
Natural vision systems are inherently incremental: new visual information is continuously incorporated while existing knowledge is preserved.

## challenge
1. it should be trainable from a stream of data in which examples of different classes occur at different times.
2. it should at any time provide a competitive multi-class classifier for the classes observed so far.
3. its computational requirements and memory footprint should remain bounded, or at least grow very slowly, with respect to the number of classes seen so far.
## existing solution and gaps
Most existing multi-class techniques simply violate i) or ii) as they can only handle a fixed number of classes and/or need all training data to be available at the same time. Naively, one could try to overcome this by training classifiers from class-incremental data streams, e.g. using stochastic gradient descent optimization. This, however, will cause the classification accuracy to quickly deteriorate, an effect known in the literature as catastrophic forgetting or catastrophic interference

The few existing techniques that do fulfill the above properties are principally limited to situations with a fixed data representation. They cannot be extended to deep architectures that learn classifiers and feature representations at the same time and are

##  proposal
1. classification by a nearest-mean-of-exemplars rule,
2. prioritized exemplar selection based on herding,
3. representation learning using knowledge distillation and prototype rehearsal.
## specific design
### Nearest-Mean-of-Exemplars Classification
To predict a label, for a new image, $x$, iCaRL uses a nearest-mean-of-exemplars classification strategy.

**Comment by swagshaw** [comment]
![image](https://user-images.githubusercontent.com/48468760/153176263-196e4b56-70a2-4431-9de1-a1a61f625409.png)

**Comment by swagshaw** [comment]
![image](https://user-images.githubusercontent.com/48468760/153176344-0ba582dc-a66f-4bc8-af97-7588b77c14cb.png)
![image](https://user-images.githubusercontent.com/48468760/153176379-c7c7b66e-f07a-4b92-bd14-0fa2be4684d5.png)
![image](https://user-images.githubusercontent.com/48468760/153176402-5f2a5c8d-f0a6-442f-bf43-3dc18a8ff1e7.png)
![image](https://user-images.githubusercontent.com/48468760/153176430-97d2fb2e-680f-4340-94ae-8dd5124f9535.png)

**Comment by swagshaw** [comment]
iCaRL是非常经典的work，典型的sampling，典型的exemplar management，再加上继承自LwF的典型的distillation loss。
